doc/quickstart.md

```diff
@@ -175,6 +174,43 @@ Python code:
 1234.5
 ```
 
+#### Structures with multiple datatypes (_read only_)
+
+TwinCAT declaration:
+```
+{attribute 'pack mode' := 1}
+TYPE sample_structure :
+STRUCT
+	rVar : LREAL;
+	rVar2 : REAL;
+	iVar : INT;
+	iVar2 : ARRAY [1..3] OF DINT;
+    sVar : STRING;
+END_STRUCT
+END_TYPE
+```
+
+Python code:
+
+First you must declare a tuple which defines the PLC structure. This should match the order
+as declared in the PLC.
+
+```python
+>>> structure_def = (
+...    ('rVar', pyads.PLCTYPE_LREAL, 1),
+...    ('rVar2', pyads.PLCTYPE_REAL, 1),
+...    ('iVar', pyads.PLCTYPE_INT, 1),
+...    ('iVar2', pyads.PLCTYPE_DINT, 3),
+...    ('sVar', pyads.PLCTYPE_STRING, 1))
+```
+
+Then you can read the structure:
+
+```python
+>>> plc.read_structure_by_name('global.sample_structure', structure_def)
+[11.1, 22.2, 3, 4, 44, 444, 'abc']
```

Comment:
As it is named read_structure_by_name I think it would make more sense to return a dictionary or a named tuple with the defined member names. What do you think?

Reply:
My initial instinct when I first starting thinking of this was to return a dictionary with the key names being the variable names as defined in the 'structure_def' as it does seem more intuiative for a structure. However I decided for the draft to go with a list to keep the return consistent with the return forms of the other functions.

I also thought that if I create a 'write_structure_by_name' method it would be easier to pass a list in to write, rather than try to interpret a dictionary?

I'm not sure as a user if they would prefer to deal with lists and then use the indexes of the list (as they know the structure definition) to get the variable or if they would prefer to deal with dictionary. Which one do you think would be easier for users to deal with in their program? I guess a dictionary makes sense if you were to not use all the variables that are in the structure as you can just pick the ones you want?

I am also still working on further tests to keep the coverage up and shall also aim to implement the ability to optionally pass the handle in like my other PR and also I believe it makes sense to be able to optionally pass the structure size in too, so you can calculate this seperately then use it every time instead of calculating all the time.

Reply:
Also what do you think the ouput of the dictionary would look like for arrays and arrays of structures?

I can imagine something like this for arrays:

```python
{
'bVar': True,
'rVar': 123.456
'iVar': 45
'varName': varValue,
'arrayVar': [1, 2, 3, 4]
}
```

And I guess you would just have a list of dictionaries if you were trying to read an array of dictionaries.

It is also easier to just pass back a list as the way the intended read from the PLC is working is that you just get back a giant list of bytes from the PLC, which is easy to turn into a list, so I think I would have to go through a list of variables anyway when manipulating the bytes into a dictionary?

Reply:
Sorry to bombard you with comments but another potential implememntation I was thinking of would be to do just amend the read_by_name method to do something like:

constants.py
```python
PLCTYPE_STRUCT = PLCTYPE_BYTE
```

ads.py
```python
def read_by_name(name, type, structure_def=None):
    if structure_def is not None:
        do stuff similar to read_structure_by_name implementation
```

users script
```python
structure_def = (
(bla, bla , bla)
)

values = plc.read_by_name('global.samplestruct', pyads.PLCTYPE_STRUCT, structure_def=structure_def)
```
returns
```python
>>>print(values)
[1,2,True, 3.456]
```

Not sure if this would be prefrable to having another standalone method?

Reply:
As the structure_def in your implementation contains fieldnames it only makes sense to return a dictionary. I don't see why the fieldnames would be needed otherwise. Also I find it more consistent to return a struct as a dict.

Maybe we could use an OrderedDict here. So the user has both possibilities:

    ordered_dict["bVar"] will return the value of a specific field
    list(ordered_dict.values()) will return all values as a list.

What do you think?

Reply:
Yes the field names in my current implementation are not used, I only added them thinking it would be easier to keep track of the structure definition as a user. I did think of an ordered dict but was trying to avoid another import. However I do think it makes sense as a dictionary as you say and this was my first thought.

So use an ordered dict as the output for a read_structure_by_name method? 

If you agree I'll change to this.

Reply:
I think an OrderedDict would be the right choice here. I don' t  mind the extra import because it is from the standard library so we won't add any new dependencies. So, yes... would be great if you could adjust the implementation to use OrderedDict.